src/pages/Detail/Detail.jsx

```diff
@@ -1,7 +1,141 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import variables from '../../styles/variables';
+import DetailNav from './component/DetailNav';
+import DetailIntro from './component/DetailIntro';
+import DetailAside from './component/DetailAside';
 
-function Detail(props) {
-  return <div>Detail</div>;
+function Detail() {
+  const [lecture, setLecture] = useState({});
+  const [isLogin, setIsLogin] = useState(false);
+  // const [subscribe, setSubscribe] = useState(false);
+  const { coverImage, coverGallery, review } = lecture;
+
+  useEffect(() => {
+    fetch(`/data/data_detail.json`)
+      .then(response => response.json())
+      .then(result => {
+        if (result.data) {
+          setLecture(result.data);
+        }
+      });
+  }, []);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      setIsLogin(true);
+    }
+  }, [localStorage.getItem('token')]);
```

Comment:
```suggestion
  const token = localStorage.getItem('token');

  useEffect(() => {
    if (token) return setIsLogin(true);
  }, [token]);
```

token을 담는 변수를 밖에서 선언하고 useEffect에 위와 같은 방식으로 적용하는 것이 가독성 측면에서 더 좋습니다
그리고 조건문 안에 한줄의 코드만 있다면 위와 같은 방식으로 early return 패턴을 적용하는 방법도 있습니다!